src/Project/Project.jl

```diff
@@ -0,0 +1,427 @@
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ --- End License
+=#
+
+#=
+    The Project is the controller in an MVC paradigm. It manages the model,
+    stored in the projectDictionary, and plotting data, and responds to enableNotifications
+    of changes.
+=#
+mutable struct Project
+    name::String
+    projectDirectory:: String
+    projectDictionary::Dict{String,Any}
+    plt::Any #For the plot
+    plotOptions::Int # = combinations of MODEL, GRID, MESH
+#
+#   For drawing
+#
+    outerBndryPoints       ::Array{Any}              # CHAIN
+    outerBndryNames        ::Array{String}
+    innerBoundaryPoints    ::Array{Any}              # Array of CHAINs
+    innerBoundaryNames     ::Array{Any}              # Array of CHAINs of names
+    innerBoundaryChainNames::Array{String}           # Array of the names of the CHAINs
+    refinementRegionPoints ::Array{Array{Float64,2}} # Array of Array of points
+    refinementRegionNames  ::Array{String}
+    refinementRegionLoc    ::Array{Array{Float64}}   # Center point of a refinement region
+    bounds                 ::Array{Float64}
+    userBounds             ::Array{Float64}
+    xGrid                  ::Array{Float64}
+    yGrid                  ::Array{Float64}
+    xMesh                  ::Array{Float64}
+    yMesh                  ::Array{Float64}
+    backgroundGridShouldUpdate::Bool
+    meshShouldUpdate        ::Bool
+end
+
+const defaultPlotPts  = 50
+const meshFileFormats = Set(["ISM", "ISM-V2", "ABAQUS"])
+const plotFileFormats = Set(["sem", "skeleton"])
+const smootherTypes   = Set(["LinearSpring", "LinearAndCrossbarSpring"])
+const statusValues    = Set(["ON", "OFF"])
+const refinementTypes = Set(["smooth", "sharp"])
+
+
+"""
+    openProject(fileName::String, folder::String)
+
+Open existing project described in the control File.
+
+    folder   = folder the control file is in
+    fileName = the name of the file
+"""
+function openProject(fileName::String, folder::String)
+
+    controlFile = joinpath(folder,fileName)
+    controlDict = ImportControlFile(controlFile)
+
+    # Strip off the file path into a temporary name
+    tempName = splitdir(fileName)
+    # Separate the project name from `.control`
+    splitName = split(tempName[2], ".")
+    # Pull the correct project name
+    s = string(splitName[1]) # This is dumb
+
+    # Instantiate an empty project with some defaults
+    proj = newProject(s,folder)
+#
+#   Overwrite defaults
+#
+    proj.projectDictionary = controlDict
+
+    assemblePlotArrays(proj)
+    clearUndoRedo()
+
+    return proj
+end
+
+
+"""
+    saveProject(proj::Project)
+
+    proj     = Project to be saved
+Save a project dictionary to the file path specified when the project was created.
+"""
+function saveProject(proj::Project)
```

Comment:
Maybe just
```suggestion
function save(proj::Project)
```
?

Reply:
Nope. This has verb+noun. I once thought it was cool that I could overload names so that I only needed a single command, like this. I stopped after a while because it made the code harder to debug.